GMG/GMG/UI/Components/Waveform/WaveformUseView.swift

```diff
@@ -0,0 +1,167 @@
+//  Copyright © 2025 ADA 4th GMG. All rights reserved.
+
+import SwiftUI
+import AVFoundation
+
+struct WaveformUseView: View {
+    @State private var amps: [Float] = []
+    @State private var progress: Double = 0
+    @State private var player: AVAudioPlayer?
+    @State private var timer: Timer?
+
+    private let bins = 150
+
+    var body: some View {
+        VStack(spacing: 16) {
+            Waveform(amps: amps, progress: progress)
+            
+            ProgressSlider(progress: $progress)
+            
+            SampleLoadButton(amps: $amps, bins: bins)
+            
+            PlaySection(isEnabled: !amps.isEmpty) {
+                playFromStart()
+                
+            }
+        }
+        .padding()
+    }
+}
+
+private extension WaveformUseView {
+    func playFromStart() {
+        progress = 0
+        timer?.invalidate()
+
+        guard let url = Bundle.main.url(forResource: "sample", withExtension: "m4a") else { return }
+        do {
+            player = try AVAudioPlayer(contentsOf: url)
+            player?.prepareToPlay()
+            player?.play()
+        } catch {
+            print("재생 실패:", error)
+            return
+        }
+
+        timer = Timer.scheduledTimer(withTimeInterval: 0.02, repeats: true) { t in
+            guard let p = player else {
+                t.invalidate()
+                return
+            }
+            if p.isPlaying {
+                progress = p.duration > 0 ? p.currentTime / p.duration : 0
+            } else {
+                t.invalidate()
+            }
+        }
+    }
+
+    func highlightRange(progress: Double, totalBins: Int, windowBins: Int) -> Range<Int> {
+        guard totalBins > 0 else { return 0..<0 }
+        let idx = Int(round(progress * Double(max(0, totalBins - 1))))
+        let half = windowBins / 2
+        let lower = max(0, idx - half)
+        let upper = min(totalBins, lower + windowBins)
+        return lower..<upper
+    }
+}
+
+extension WaveformUseView {
+
+    struct Waveform: View {
+        let amps: [Float]
+        let progress: Double
+
+        var body: some View {
+            ZStack {
+                Image("WaveformBackground")
+                    .resizable()
+                    .scaledToFit()
+                    .background(Color.white)
+                    .clipShape(RoundedCornerRectangle(radius: 20, corners: [.bottomLeft, .bottomRight]))
```

Comment:
내장 Shape로도 구현할 수 있어요!

```suggestion
                    .clipShape(UnevenRoundedRectangle(bottomLeadingRadius: 20, bottomTrailingRadius: 20))
```